.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,188 @@
+          echo "::set-output name=version::${VERSION}"
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [supported-versions-matrix, dependency-versions-matrix]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version: ${{ fromJson(needs.supported-versions-matrix.outputs.version) }}
+        deps-version: ${{ fromJson(needs.dependency-versions-matrix.outputs.version) }}
+        include:
+          - php-version: "8.0"
+            composer-args: "--ansi --no-interaction --no-progress --prefer-dist --ignore-platform-reqs"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: mbstring, intl, readline
+          ini-values: memory_limit=-1
+          coverage: none
+          tools: composer:v2, cs2pr
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
+          key: "php-${{ matrix.php-version }}-${{ matrix.deps-version }}-${{ hashFiles('**/composer.lock') || github.ref }}"
+          restore-keys: "php-${{ matrix.php-version }}-${{ matrix.deps-version }}-${{ hashFiles('**/composer.lock') || github.ref }}"
+
+      - name: "Install dependencies (lowest)"
+        if: matrix.deps-version == 'lowest'
+        env:
+          COMPOSER_ARGS: ${{ matrix.composer-args || '--ansi --no-interaction --no-progress --prefer-dist' }}
+          COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          composer update ${COMPOSER_ARGS} --prefer-lowest
+          composer show --ansi
+
+      - name: "Install dependencies (locked)"
+        if: matrix.deps-version == 'locked'
+        env:
+          COMPOSER_ARGS: ${{ matrix.composer-args || '--ansi --no-interaction --no-progress --prefer-dist' }}
+          COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          composer install ${COMPOSER_ARGS}
+          composer show --ansi
+
+      - name: "Install dependencies (latest)"
+        if: matrix.deps-version == 'latest'
+        env:
+          COMPOSER_ARGS: ${{ matrix.composer-args || '--ansi --no-interaction --no-progress --prefer-dist' }}
+          COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          composer update ${COMPOSER_ARGS}
+          composer show --ansi
+
+      - name: Setup problem matcher for PHPUnit
+        run: echo "::add-matcher${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Run unit tests on PHP ${{ matrix.php-version }}
+        run: "composer test"
+
+  checks:
+    name: Checks
+    runs-on: ubuntu-latest
+    needs: [supported-checks-matrix, latest-supported-php-7]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        check: ${{ fromJson(needs.supported-checks-matrix.outputs.check) }}
+        exclude:
+          - check: test
```

Comment:
I _think_ we might be able to use expressions here; if so, we could maybe make an exclusion list for determining which checks to exclude, instead of which ones to include.